tests/test_wsireader.py

```diff
@@ -2319,6 +2325,89 @@ def test_ngff_inconsistent_multiscales_versions(tmp_path, caplog, remote_sample)
     assert "multiple versions" in caplog.text
 
 
+def test_jp2_no_mpp_appmag(tmp_path):
+    """Test WSIReader init for a JP2 with no resolution metadata."""
+    path = tmp_path / "test.jp2"
+    glymur.Jp2k(
+        path,
+        data=np.ones((128, 128, 3), np.uint8),
+    )
+    _ = WSIReader.open(path).info
+
+
+def test_jp2_empty_xml(tmp_path):
+    """Test WSIReader init for a JP2 with empty Jp2 XML."""
+    path = tmp_path / "test.jp2"
+    jp2 = glymur.Jp2k(
+        path,
+        data=np.ones((128, 128, 3), np.uint8),
+    )
+    xml_path = tmp_path / "data.xml"
+    xml_path.write_text("<Jp2></Jp2>")
+    xmlbox = glymur.jp2box.XMLBox(filename=xml_path)
+    jp2.append(xmlbox)
+    _ = WSIReader.open(path).info
+
+
+def test_jp2_empty_xml_empty_description(tmp_path):
+    """Test WSIReader init for a JP2 with an empty Jp2 XML description tag."""
```

Comment:
Is this JP2 or Jp2? It would be good to be consistent.

Reply:
It's JP2

Reply:
Should it be JP2 XML?

Reply:
Yep